buddy_alloc.h

```diff
@@ -1782,9 +1988,9 @@ static void bitset_shift_left(unsigned char *bitset, size_t from_pos, size_t to_
 }
 
 static void bitset_shift_right(unsigned char *bitset, size_t from_pos, size_t to_pos, size_t by) {
-    ssize_t length = to_pos - from_pos;
+    ssize_t length = (ssize_t)(to_pos - from_pos);
     while (length >= 0) {
-        size_t at = from_pos + length;
+        size_t at = from_pos + (size_t)length;
```

Comment:
since length >=0 is checked in while, we can convert length to size_t without problem.

Reply:
What's the value here, beyond the compiler warning? It doesn't change the behavior of the code.

Reply:
It won't. Most of the warning is about implicit conversion. I know this is more of a C thing. Coming from Go and Rust, most of the signed to unsigned and vice verse conversion are explicit and required to specify compiler. I personally like explicit rather than implicit.

Reply:
Fair enough. I'll fixed the coverage thing and I'll work on adding some of the options that you suggested to the default build. I'll keep you posted.

Meanwhile if you have any issues with the allocator or need any new functionality - feel free to file issues or even send in PRs.